plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/webhooks/WebHooksComponent.java

```diff
@@ -0,0 +1,216 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.webhooks;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.eclipse.swt.dnd.Transfer;
+import org.eclipse.swt.events.MouseAdapter;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Text;
+import org.jboss.tools.foundation.ui.util.BrowserUtility;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.internal.ui.OpenShiftImages;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+import com.openshift.restclient.model.IBuildConfig;
+import com.openshift.restclient.model.build.BuildTriggerType;
+import com.openshift.restclient.model.build.IBuildTrigger;
+import com.openshift.restclient.model.build.IWebhookTrigger;
+
+/**
+ * @author Fred Bricon
+ * @author Andre Dietisheim
+ */
+public class WebHooksComponent extends Composite {
+
+	private IBuildConfig buildConfig;
+
+	public WebHooksComponent(IBuildConfig buildConfig, Composite parent, int style) {
+		super(parent, style);
+		this.buildConfig = buildConfig;
+		
+		createControls(buildConfig, parent);
+	}
+
+	private void createControls(IBuildConfig buildConfig, Composite parent) {
+		GridLayoutFactory.fillDefaults()
+			.applyTo(this);
+
+		Label webhookExplanation = new Label(this, SWT.WRAP);
+		webhookExplanation.setText("Webhook triggers allow you to trigger a new build by sending a request to the OpenShift API endpoint. "
```

Comment:
Im also wondering if it makes sense to add in a help link to: https://docs.openshift.org/latest/dev_guide/builds.html#build-triggers.  The generic request, it might be useful if a user knew they could provide a payload